pyne/vta.py

```diff
@@ -0,0 +1,551 @@
+    v : list
+        Direction vector from start to end.
+
+    Returns:
+    --------
+    x_idx : int
+        Modified x_idx.
+    y_idx : int
+        Modified y_idx.
+    z_idx : int
+        Modified z_idx.
+    tp : Point
+        Point after moving along the direction of v.
+    t_temp : float
+        Modifed travel length from the start.
+    """
+    t_min = min(t_maxs)
+    if t_maxs[0] == t_min:
+        x_idx += 1
+    elif t_maxs[1] == t_min:
+        y_idx += 1
+    elif t_maxs[2] == t_min:
+        z_idx += 1
+    tp.x += t_min * v[0]
+    tp.y += t_min * v[1]
+    tp.z += t_min * v[2]
+    t_temp += t_min
+    return x_idx, y_idx, z_idx, tp, t_temp
+
+def _move_to_boundary(tp, t_min, v):
+    """
+    Move the point to the nearest boundary when Initialize the problem.
+
+    Parameters:
+    -----------
+    tp : Point
+        Temporary point. Actually, it is also the start point.
+    t_min : float
+        Distance to the nearest (positive) boundary. For a 3D mesh, there are 6
+        outside boundaries. Therefore there are 6 value of the distance from the
+        start to the boundaries. This t_min is the smallest one among the
+        positive distance. (The distance is negtive when the direction is
+        negtive.)
+    v : list
+        Direction vector.
+
+    Return:
+    -------
+    tp : Point
+        Modified point. This point lies on the nearest boundary. Therefore, we
+        moved point from outside the mesh to the mesh boundary (or the extension
+        of the boundary, which still outside the mesh).
+    """
+
+    # calculate the new coordinate
+    tp.x += t_min * v[0]
+    tp.y += t_min * v[1]
+    return tp
+
+def _ray_voxel_traverse(bounds, start, end):
+    """
+    Voxel traversal algorithm.
+    Return the voxels that intersect with the line segment.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries of the mesh grid.
+    start : Point
+        Start point.
+    end : Point
+        End point.
+
+    Return:
+    -------
+    voxles : list
+        List of voxel indices. These voxels intersect with the line segment.
+    """
+    # Initialization
+    # Find the voxel that start point in
+    tp = Point(start.x, start.y, start.z)
+    x_idx, y_idx, z_idx = -1, -1, -1
+    voxel_list = []
+    v = _cal_dir_v(start, end)
+    t_end = _distance(start, end)
+    t_temp = 0.0
+    while t_temp < t_end:
+        if _is_point_in_mesh(bounds, tp):
+            # the point is in the mesh
+            # calculate the voxel id
+            x_idx = _find_voxel_idx_1d(bounds[0], tp.x, v[0])
+            y_idx = _find_voxel_idx_1d(bounds[1], tp.y, v[1])
+            z_idx = _find_voxel_idx_1d(bounds[2], tp.z, v[2])
+            if -1 in (x_idx, y_idx, z_idx):
+                # point outside the mesh
+                return voxel_list
+            # add current voxel
+            voxel_list.append([x_idx, y_idx, z_idx])
```

Comment:
It might be worth considering tuples here. You'll be looping over these values later (presumably) and won't want them to accidentally get changed.